test/functional/apps/dashboard_elements/links/links_navigation.ts

```diff
@@ -214,9 +214,7 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
         expect(isDisabled).to.be('true');
       });
 
-      // TODO We should not be using an external website for our tests. This will be flaky
-      // if external network connectivity issues exist.
```

Comment:
@nickpeihl You added this comment in https://github.com/elastic/kibana/pull/167896, but there was no further context. What issues would cause this to to be flaky? We aren't checking the **contents** of the external URL, we are just ensuring that navigation happens as expected (i.e. verifying that the URL matches our expectations). `https://example.com/` is also fully safe to navigate to, since it is a URL that is reserved for testing purposes 🤔 Do you recall the issues here?

Reply:
I don't fully recall if there was a documented reason to avoid using external resources in tests. CI can run in numerous environments which may affect loading external resources. That said, you may be correct that the test would still pass even if https://example.com can not be loaded. 

But now I think these functional tests could be removed in favor of the [unit tests](https://github.com/elastic/kibana/blob/main/src/platform/plugins/private/links/public/components/external_link/external_link_component.test.tsx) which are cheaper and test the same features.

Reply:
We could definitely replace some of these tests with unit tests! But is it okay if we merge this as-is for now?

Reply:
We already have unit tests for opening in a new tab, so I don't think we need to create any new tests. But I'm OK with merging this to unblock.